### DISCUSSION TITLE
Customize registration form + Invitations

### DISCUSSION THREAD

**Discussion by kolisko** [keycloak/keycloak#27801] [Q&A]
Hello,

I have two questions. The first concerns modifying the registration form. Currently, the form by default shows items such as first name, last name, email, password (and the fields are mandatory). Is it possible to modify these items through the admin interface? I would like the registration not to require a first name and last name. Can this be set up through KC configuration, or is it necessary to edit the .ftl files manually and intervene in the template modification? I would prefer to avoid this if possible.

The second question concerns the possibility of sending invitations to users. Currently, when a user registers through the registration form and it is set to verify the email address, the user first receives a verification link by email, which, when confirmed, then grants access - so it is a one-step process with a click on the link in the email. I would like a similar functionality (one-step flow) for invitations. That is, I would like to register new users who would receive the same verification link in their email. I tried to use the send-verify-email API for this purpose, but the email template differs from the one triggered by registration through the form, and upon opening the link - I see a multi-step screen flow (email verification request, confirmation, etc.). This verification is different from what happens when sending the registration form. I would like to ask if it is possible to trigger the same action for email verification through the API as is triggered by registration through the standard form?

Thanks, M

**Comment by kolisko** [comment]
I note that I have Keycloak version `22.0.5` and in it, I do not see the option to enable the user profile, nor do I see a "User Profile" tab, as in version 24, where it is possible to easily remove these two attributes (firstName, lastName).

**Comment by bitbythecron** [comment]
I second the motion for an Invitations feature. This is a critical part of the registration process for many apps, where an admin needs to invite you to come use the system. Is this page the only place where an Invitations feature is being tracked, or is there a Roadmap document showing when/if this would ever be added?

I can make do by usig the Keycloak REST API for now to create users (with temp passwords), send my own emails, etc. But a built-in Invitation feature would be awesome.
